helix-core/src/main/java/org/apache/helix/controller/stages/CustomizedViewAggregationStage.java

```diff
@@ -119,6 +117,7 @@ public void execute(final ClusterEvent event) throws Exception {
               "Failed to calculate customized view for resource " + resource.getResourceName(), ex);
         }
       }
+      updatedCustomizedViews.clear();
```

Comment:
Why? This list is a local variable, right?

Reply:
It's a variable defined outside state type for loop. If we do not clear it, it actually contains resource -> customized view mapping for all state types. When we update ZK, there's problem as we are updating state by state. 
But I actually found this change is ugly, so I just moved the list inside the loop.